File: solutions/csharp/flatten-array/1/FlattenArray.cs
using System;
using System.Collections;

public static class FlattenArray
{
    public static IEnumerable Flatten(IEnumerable input)
    {
        foreach (var item in input)
        {
            if (item is IEnumerable enumerable)
            {
                foreach (var subitem in Flatten(enumerable))
                    yield return subitem;
            }
            else if (item != null)
            {
                yield return item;
            }
        }

    }
}

File: solutions/python/flatten-array/1/flatten_array.py
from collections.abc import Iterable

def flatten(iterable):
    return list(_flatten(iterable))

def _flatten(value):
    if isinstance(value, Iterable):
        for sub_value in value:
            for item in _flatten(sub_value):
                yield item
    elif value is not None:
        yield value
